### ISSUE TITLE
AutoMapper 5.0.2. Injection with Autofac

### ISSUE THREAD

**Issue by RobvanGeloven** [AutoMapper/AutoMapper#1524]
Since upgrading to AutoMapper 5.0.2. I've experienced issues with Autofac. The code mentioned in this issue: https://github.com/AutoMapper/AutoMapper/issues/1109 
works fine in the 4.X branch but gives me the error "Mapper not initialized. Call Initialize with appropriate configuration." when using 5.0.2.

**Comment by RobvanGeloven** [comment]
This would be the relevant code:
```cs
        builder.Register(context => new MapperConfiguration(configuration =>
        {
            foreach (var profile in context.Resolve<IEnumerable<Profile>>())
            {
                configuration.AddProfile(profile);
            }
        }))
            .AsSelf()
            .SingleInstance();

        builder.Register(context => context.Resolve<MapperConfiguration>()
            .CreateMapper(context.Resolve))
            .As<IMapper>()
            .InstancePerLifetimeScope();
```

I get the error the moment I try to call a Map<> in the code

**Comment by jbogard** [comment]
How are you calling Map in the code?

**Comment by RobvanGeloven** [comment]
Ah. Found the 'culprit'. When using ProjectTo it fails. When calling the 'Map' method everything is in order. Missed that one....

**Comment by jbogard** [comment]
The reason I ask is that those exceptions are *only* thrown for static access.

**Comment by jbogard** [comment]
Ah, yes, ProjectTo has a special overload for using instance-based versions of AutoMapper. Unfortunately it's an extension method, so you have to pass in the configuration object, no injection supported :(

**Comment by RobvanGeloven** [comment]
Aah :) Then as far as I'm concerned not a bug. Closing this!

**Comment by jbogard** [comment]
The exception message leaves something to be desired, how does this look?

Mapper not initialized. Call Initialize with appropriate configuration. If you are trying to use mapper instances through a container or otherwise, make sure you do not have any calls to the static Mapper.Map methods, and if you're using ProjectTo or UseAsDataSource extension methods, make sure you pass in the appropriate IConfigurationProvider instance.

**Comment by RobvanGeloven** [comment]
Sounds good :+1:

**Comment by jbogard** [comment]
Fixed in #1527
